### DISCUSSION TITLE
Issue with OpenLog Escape Command on MacOS (ASCII 26 Not Recognized)

### DISCUSSION THREAD

**Discussion by angbas06** [community/community#141523] [Programming Help] [answered]
### Body

I am encountering an issue with the OpenLog device. I’m using MacOS and have connected the OpenLog via an FTDI adapter, accessing the serial interface through the screen command. The device is logging data correctly, but the escape command, which should be sent as ASCII 26 three times (Ctrl+Z x3), is not being recognized. This prevents me from accessing the microSD card to view the logged files.

I have followed the instructions provided in the guide, but despite my attempts, the escape command doesn’t work. Has anyone encountered the same problem or knows how to solve it?

### Guidelines

- [X] I have read and understood this category's [guidelines](https://github.com/orgs/community/discussions/51384) before making this post.

**Comment by tpmst** [comment]
Hello @angbas06 

the first thing that poped into my head where the Baud Rate and Connection Settings. 
Ensure that the baud rate set in your terminal matches the baud rate expected by OpenLog (commonly 9600 bps by default).
Verify that the settings (parity, stop bits, etc.) are correctly configured according to the OpenLog specifications.


You could also try to use a different Terminal Programm, which is better suited.
While `screen` is commonly used, it might not handle sending control characters in the way you need. Alternative terminal programs like `minicom` or `CoolTerm` might offer better control over serial communication and handling of special characters.
 - Minicom: You can install minicom on MacOS using Homebrew with the command brew install minicom. Once installed, you can connect to the OpenLog with minicom -D /dev/tty.usbserial-XXXX -b 9600 where /dev/tty.usbserial-XXXX is your FTDI device.
 - CoolTerm: A graphical option that might give more intuitive control over serial connections and settings.

**Reply by angbas06** [reply]
Hello, I have already tried both methods. I tested the config with both 115200 and 9600, but nothing changes. The command is not being registered, and the mode of the OpenLog doesn't change either. In the config.txt, I have this:

115200,26,3,0,1,1,0
baud,escape,esc#,mode,verb,echo,ignoreRX

I have no idea what the issue could be. I'm concerned it might be related to the Mac, but I also tried it on a Raspberry Pi, and still nothing changed. My next step is to try it on a Windows computer to rule out any OS-related problems.

**Reply by tpmst** [reply]
I dont think its an issue with your OS, it should be doable on MAC aswell.

Sometimes the issue might be related to when the command is sent. Ensure the OpenLog is not in the middle of writing to the SD card. Wait a few seconds after the last data write before sending the escape command.

**Reply by tpmst** [reply]
Wait i have an other Idea start `screen` with the explicit command to interpret Ctrl+Z as SIGTSTP (stop signal), use `screen /dev/tty.usbserial-XXXX 9600`.
You can also try to send the Ctrl+Z character manually by typing `Ctrl+V` followed by `Ctrl+Z` in the terminal to ensure it's passed literally to the device.

**Reply by angbas06** [reply]
Thank you for your help, but could you please clarify what you mean? Are you suggesting that I copy the echo command equivalent to Ctrl+Z and send it?

**Reply by tpmst** [reply, marked as answer]
Yes, exactly! When I suggested using the echo command or manually sending the Ctrl+Z character, the idea is to ensure that the exact character sequence required by OpenLog to trigger the escape command is being sent correctly

Open Terminal and connect to your OpenLog via screen with the appropriate device identifier and baud rate. For example:
```
screen /dev/tty.usbserial-XXXX 9600
```
Replace /dev/tty.usbserial-XXXX with the actual identifier for your FTDI or serial adapter as it appears in /dev/.

**Manually:**
Once connected via screen, to send the Ctrl+Z character literally:
First, press Ctrl-V which tells the terminal to treat the following keystrokes literally.
Then press Ctrl-Z.
This combination ensures that Ctrl+Z is sent as an input character to OpenLog and not interpreted as a command by the terminal or screen.

**Command Line (outside of screen):**
If you prefer or need to script the sending of Ctrl+Z, you can use the echo command outside of screen by writing directly to the device. For example:
```
echo -e '\x1a\x1a\x1a' > /dev/tty.usbserial-XXXX
```
Replace /dev/tty.usbserial-XXXX with your actual device path. This sends the Ctrl+Z character three times directly to OpenLog.

**Reply by angbas06** [reply]
Ok, I think we're getting closer. Things are behaving a bit differently now. Copy-pasting (ctrl-v) didn't help much, but sending the commands externally did. However, I've encountered a new issue. Normally, when the OpenLog is recording, a LED lights up for each individual character, and when a second character is typed, the LED turns off. Now, after I sent three characters externally outside of the screen command, the terminal froze, and I couldn’t type anything anymore (not even getting a command prompt). When I opened a second terminal and ran the screen command, the LED on the OpenLog stopped lighting up and turning off as expected.

**Reply by tpmst** [reply]
If the terminal becomes unresponsive after sending commands, ensure that the serial port is not being left open by a process. You might want to check which processes are accessing the serial port. On macOS, you can check this using the Terminal:
```
lsof | grep tty.usbserial
```

Consider writing a small script that opens the serial port, sends the necessary commands, and then closes the port properly.

**Reply by tpmst** [reply]
Hello @angbas06 

did it work?

**Reply by angbas06** [reply]
Hi @tpmst , sorry for going silent yesterday. I'll be at work shortly and will give it a try. Thanks so much for your interest!
